go.mod

```diff
@@ -25,45 +26,36 @@ require (
 	github.com/imdario/mergo v0.3.9 // indirect
 	github.com/jinzhu/gorm v1.9.12
 	github.com/jmoiron/sqlx v1.2.0
+	github.com/juju/testing v0.0.0-20210302031854-2c7ee8570c07 // indirect
 	github.com/julienschmidt/httprouter v1.3.0
-	github.com/kr/logfmt v0.0.0-20140226030751-b84e30acd515
 	github.com/labstack/echo v3.3.10+incompatible
-	github.com/labstack/gommon v0.3.0 // indirect
+	github.com/labstack/echo/v4 v4.2.1
 	github.com/lib/pq v1.2.0
 	github.com/mailru/easyjson v0.7.1
 	github.com/mattn/go-sqlite3 v2.0.3+incompatible
 	github.com/miekg/dns v1.1.29
 	github.com/opentracing/opentracing-go v1.1.0
-	github.com/philhofer/fwd v1.0.0
+	github.com/philhofer/fwd v1.0.0 // indirect
 	github.com/pierrec/lz4 v2.5.1+incompatible // indirect
-	github.com/pmezard/go-difflib v1.0.0
 	github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0 // indirect
 	github.com/signalfx/golib v2.4.0+incompatible
 	github.com/stretchr/testify v1.5.1
 	github.com/syndtr/goleveldb v1.0.0
-	github.com/tebeka/go2xunit v1.4.10
 	github.com/tidwall/buntdb v1.1.2
 	github.com/tinylib/msgp v1.1.0
 	go.mongodb.org/mongo-driver v1.3.2
-	golang.org/x/net v0.0.0-20200202094626-16171245cfb2
+	golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
-	golang.org/x/sys v0.0.0-20200116001909-b77594299b42
+	golang.org/x/sys v0.0.0-20201119102817-f84b799fce68
 	google.golang.org/api v0.21.0
 	google.golang.org/grpc v1.28.1
-	gopkg.in/Shopify/sarama.v1 v1.20.1
+	gopkg.in/Shopify/sarama.v1 v1.26.1
 	gopkg.in/olivere/elastic.v3 v3.0.75
 	gopkg.in/olivere/elastic.v5 v5.0.85
-	gopkg.in/yaml.v2 v2.2.8
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.0.0-20190819141258-3544db3b9e44
 	k8s.io/apimachinery v0.0.0-20190817020851-f2f3a405f61d
 	k8s.io/client-go v0.0.0-20190819141724-e14f31a72a77
 	k8s.io/utils v0.0.0-20200414100711-2df71ebbae66 // indirect
 )
```

Comment:
this has lots of libraries and frameworks listed as deps. I don't thin the tracer should pull all these in. I suspect they are coming in because of instrumentations. Shouldn't each instrumentation be it's own Go module with it's deps instead of tracer pulling all the deps?

Reply:
This was fine when using go.mod only in CI but for real apps it might be very inconvenient and annoying if a simple tracer pulls in all these frameworks and libraries. I think each instrumentation should have it's own go.mod and go.sum file.

Reply:
+1 on having each instrumentation package be its own module to limit dependencies for users to only what they need.

Reply:
The dependencies are strange. they appeared when I invoked `go mod tidy`.
`go mod why` does not tell anything. quickly looked at the `go mod graph` results but I need to look at it next week.

I think your concerns are valid. I will try to test how a `go.mod` and `go.sum` will like if someone uses the tracer. If it will be big.

I will try to make a separate PR that will only tackle the problems with `contrib/mongodb/mongo-go-driver/mongo/mongo_test.go`. So we can solve https://github.com/signalfx/signalfx-go-tracing/issues/49 without migrating to Go Modules in a potentially ugly way.

Reply:
https://github.com/signalfx/signalfx-go-tracing/pull/65

Reply:
I just want to confirm that having no `go.mod` at all vs having a big one in the root is probably better.
You can find the differences from the "consumer" POV here: https://github.com/pellared/signalfx-go-tracing-test/pull/1 
I will try to make a new PR to make a Go Modules for each instrumentation.